chalk-integration/src/program.rs

```diff
@@ -42,6 +42,12 @@ pub struct Program {
     pub associated_ty_values:
         BTreeMap<AssociatedTyValueId<ChalkIr>, Arc<AssociatedTyValue<ChalkIr>>>,
 
+    // From opaque type name to item-id. Used during lowering only.
+    pub opaque_ty_ids: BTreeMap<Identifier, OpaqueTyId<ChalkIr>>,
+
+    /// For each opaque type:
+    pub opaque_ty_data: BTreeMap<OpaqueTyId<ChalkIr>, Arc<OpaqueTyDatum<ChalkIr>>>,
```

Comment:
This can be done in a follow-up PR, but I think we should add a `opaque_names: BTreeMap<TraitId<ChalkIr>, Identifier>` or something like that, created during lowering...